modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -68,10 +73,21 @@ func New(cfg Config, registry registry.Registry) gen.Processor {
 	if cfg.Subprocessors[Size] {
 		p.spanMetricsSizeTotal = registry.NewCounter(metricSizeTotal, labels)
 	}
+
+	filter, err := spanfilter.NewSpanFilter(cfg.FilterPolicies)
+	if err != nil {
+		return nil, err
+	}
+
+	if filter != nil {
```

Comment:
based on this it look like you expect that filter can sometimes be nil, but below 

```
  if p.filter.ApplyFilterPolicy(rs.Resource, span) {
```

you don't test for nil. is there a mismatch here?

Reply:
This has been refactored.